akvo/rsr/spa/app/modules/editor/reducer.js

```diff
@@ -281,6 +275,28 @@ export default (state = initialState, action) => {
     case actionTypes.VALIDATION_SYNC:
       newState[sectionKey].errors = validateSection(sectionKey, state.validations, newState[sectionKey].fields)
       return newState
+    case actionTypes.SET_EXTERNAL_PROJECT:
+      return { ...state, externalProjects: action.payload }
+    case actionTypes.ADD_EXTERNAL_PROJECT:
+      return { ...state, saving: true, externalProjects: [...state.externalProjects, action.payload] }
+    case actionTypes.REMOVE_EXTERNAL_PROJECT:
+      return { ...state, saving: true, externalProjects: state.externalProjects.filter((p) => p?.id !== action?.payload?.id) }
+    case actionTypes.SET_PARENT_PROJECT:
+      if (action.payload) {
+        const hasParent = ((state?.section1?.fields?.uuid) && (action.payload.uuid !== state.section1.fields.uuid))
+        return {
+          ...state,
+          section1: {
+            ...state?.section1,
+            fields: {
+              ...state?.section1?.fields,
+              contributesToProject: (!state?.section1?.fields?.contributesToProject && hasParent) ? action.payload.id : state?.section1?.fields?.contributesToProject
```

Comment:
minor nitpick: this is a long line. If javascript allows it, maybe a linebreak or two would be good e.g
```js
condition ?
   trueTree : falseTree
```

Reply:
got it, Thanks @MichaelAkvo I'm working to fix it